app/middleware/stripe-setup/check-company-number-not-submitted.test.js

```diff
@@ -0,0 +1,126 @@
+'use strict'
+
+// NPM dependencies
+const chai = require('chai')
+const chaiAsPromised = require('chai-as-promised')
+const proxyquire = require('proxyquire')
+const sinon = require('sinon')
+
+// Local dependencies
+const paths = require('../../paths')
+
+// Global setup
+chai.use(chaiAsPromised)
+const { expect } = chai // must be called after chai.use(chaiAsPromised) to use "should.eventually"
+
+describe('Check "Company registration number" not submitted middleware', () => {
+  let req
+  let res
+  let next
+
+  beforeEach(() => {
+    req = {
+      correlationId: 'correlation-id',
+      account: {
+        gateway_account_id: '1'
+      },
+      flash: sinon.spy()
+    }
+    res = {
+      setHeader: sinon.stub(),
+      status: sinon.spy(),
+      render: sinon.spy(),
+      redirect: sinon.spy()
+    }
+    next = sinon.spy()
+  })
+
+  it('should call next when "VAT number" flag is false', done => {
+    const middleware = getMiddlewareWithConnectorClientResolvedPromiseMock({
+      companyNumber: false
+    })
+
+    middleware(req, res, next)
+
+    setTimeout(() => {
+      expect(next.calledOnce).to.be.true // eslint-disable-line
+      expect(req.flash.notCalled).to.be.true // eslint-disable-line
+      expect(res.redirect.notCalled).to.be.true // eslint-disable-line
+      done()
+    }, 250)
+  })
+
+  it('should redirect to the dashboard with error message when "VAT number" flag is true', done => {
```

Comment:
This test name and the one above should say "Company number"

Reply:
🤦 updated